Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs

```diff
@@ -0,0 +1,176 @@
+using Content.Server.Popups;
+using Content.Server.Cargo.Systems;
+using Content.Server.Cargo.Components;
+using Content.Server.Radio.EntitySystems;
+using Content.Server.Shipyard.Components;
+using Content.Server.Shuttles.Components;
+using Content.Server.Shuttles.Systems;
+using Content.Server.Station.Systems;
+using Content.Shared.GameTicking;
+using Content.Shared.Shipyard.Events;
+using Content.Shared.Shipyard.BUI;
+using Content.Shared.Shipyard.Prototypes;
+using Content.Shared.Shipyard.Components;
+using Content.Shared.Access.Systems;
+using Content.Shared.Access.Components;
+using Content.Shared.Shipyard;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Players;
+using Robust.Shared.Prototypes;
+using Content.Shared.Radio;
+
+namespace Content.Server.Shipyard.Systems
+{
+    public sealed class ShipyardConsoleSystem : SharedShipyardSystem
+    {
+        [Dependency] private readonly AccessReaderSystem _access = default!;
+        [Dependency] private readonly PopupSystem _popup = default!;
+        [Dependency] private readonly UserInterfaceSystem _ui = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly ShipyardSystem _shipyard = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly CargoSystem _cargo = default!;
+        [Dependency] private readonly RadioSystem _radio = default!;
+        [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+        public void InitializeConsole()
+        {
+            SubscribeLocalEvent<ShipyardConsoleComponent, ShipyardConsolePurchaseMessage>(OnPurchaseMessage);
+            SubscribeLocalEvent<ShipyardConsoleComponent, BoundUIOpenedEvent>(OnConsoleUIOpened);
+            //SubscribeLocalEvent<ShipyardConsoleComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(Reset);
+        }
+
+        private void OnInit(EntityUid uid, SharedShipyardConsoleComponent orderConsole, ComponentInit args)
+        {
+            //_shipyard.SetupShipyard(); ///if we have to start up the shipyard from here later
+        }
+
+        private void Reset(RoundRestartCleanupEvent ev)
+        {
+            //_shipyard.Shutdown(); //round cleanup event in case of needing OnInit;
+        }
+
+        private void OnPurchaseMessage(EntityUid uid, SharedShipyardConsoleComponent component, ShipyardConsolePurchaseMessage args)
+        {
+            if (args.Session.AttachedEntity is not { Valid : true } player)
+            {
+                return;
+            }
+
+            if (TryComp<AccessReaderComponent>(uid, out var accessReaderComponent) && accessReaderComponent.Enabled && !_access.IsAllowed(player, accessReaderComponent))
```

Comment:
_access.IsAllowed on it's own should be sufficient here.

Reply:
i still need to grab the access reader component in the trycomp anyway, and this way if the console has it's access component removed through admin or I guess emag or whatever it will just skip the check.  I removed the .Enabled check however